_drafts/practical_zephyr_01_kconfig.md

```diff
@@ -0,0 +1,986 @@
+
+```bash
+tree --charset=utf-8 --dirsfirst
+.
+├── boards
+│   └── nrf52840dk_nrf52840.conf
+├── src
+│   └── main.c
+├── CMakeLists.txt
+├── prj.conf
+└── prj_release.conf
+
+$ cat boards/nrf52840dk_nrf52840.conf
+```
+```conf
+CONFIG_UART_CONSOLE=n
+```
+
+Then, we perform a _pristine_ build of the project.
+
+```bash
+$ west build --board nrf52840dk_nrf52840 -d ../build --pristine
+```
+
+In the output of `west build` shown below you can see that the new file `boards/nrf52840dk_nrf52840.conf` is indeed merged into the `Kconfig` configuration. You can also see a warning that the Zephyr library `drivers__console` is excluded from the build since it now has no `SOURCES`:
+
+```
+Parsing /opt/nordic/ncs/v2.4.0/zephyr/Kconfig
+Loaded configuration '/opt/nordic/ncs/v2.4.0/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
+Merged configuration '/path/to/01_kconfig/prj.conf'
+Merged configuration '/path/to/01_kconfig/boards/nrf52840dk_nrf52840.conf'
+Configuration saved to '/path/to/build/zephyr/.config'
+...
+CMake Warning at /opt/nordic/ncs/v2.4.0/zephyr/CMakeLists.txt:838 (message):
+  No SOURCES given to Zephyr library: drivers__console
+  Excluding target from build.
+```
+
+You can of course verify that the symbol is indeed disabled by checking the `zephyr/.config` file in the `build` directory.
+
+> **Notice:** Previous versions of Zephyr used a `prj_<board>.conf` fragment in the project root directory instead of board fragments in the `boards` directory. At the time of writing the build system still merges any such file into the configuration. This `prj_<board>.conf` fragment, however, is deprecated, and `boards/<board>.conf` should be used instead.
+
+How are board fragments used by our previously created `release` build type? Let's try it out and have a look at the build output:
+
+```bash
+$ rm -rf ../build
+$ west build --board nrf52840dk_nrf52840 -d ../build -- -DCONF_FILE=prj_release.conf
+```
+```
+Parsing /opt/nordic/ncs/v2.4.0/zephyr/Kconfig
+Loaded configuration '/opt/nordic/ncs/v2.4.0/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
+Merged configuration '/path/to/01_kconfig/prj_release.conf'
+Configuration saved to '/path/to/build/zephyr/.config'
+```
+
+Well, this is awkward: Even though we provided a board-specific `Kconfig` fragment it is ignored as soon as we switch to our alternative `prj_release.conf` configuration. This is, however, [working as intended](https://docs.zephyrproject.org/latest/build/kconfig/setting.html#the-initial-configuration): When using alternative `Kconfig` files in the format `prj_<build>.conf`, Zephyr looks for a board fragment matching the file name `boards/<board>_<build>.conf`.
```

Comment:
Really great example here, even helped me clarify how this works!

Reply:
I was actually confused and to be honest still not a big fan of this weird implicit build type :D